code/drasil-code/lib/Language/Drasil/Code/Imperative/Generator.hs

```diff
@@ -47,8 +47,8 @@ import Text.PrettyPrint.HughesPJ (isEmpty, vcat)
 -- | Initializes the generator's 'DrasilState'.
 -- 'String' parameter is a string representing the date.
 -- \['Expr'\] parameter is the sample input values provided by the user.
-generator :: Lang -> String -> [Expr] -> Choices -> CodeSpec -> DrasilState
-generator l dt sd chs spec = DrasilState {
+generator :: Lang -> String -> [Expr] -> Choices -> CodeSpec -> String -> DrasilState
+generator l dt sd chs spec vers = DrasilState {
```

Comment:
Since we're going to try out the TH approach, I don't think we need to add a parameter at all now. We can just directly refer to the version held in `drasil-metadata` wherever `vers` is currently used.

Reply:
Ah, this comment applies to everything below too where the extra option is added.